src/main/java/com/microfocus/octane/plugins/views/CoverageResource.java

```diff
@@ -47,12 +52,49 @@ public CoverageResource(UserManager userManager) {
     }
 
     @GET
-    public Response getCoverage(@Context HttpServletRequest request, @QueryParam("project-key") String projectKey, @QueryParam("issue-key") String issueKey, @QueryParam("issue-id") String issueId) {
+    public Response getCoverage(@Context HttpServletRequest request, @QueryParam("project-key") String projectKey, @QueryParam("issue-key") String issueKey, @QueryParam("issue-id") String issueId,
+                                @QueryParam("workspace-config-id") String workspaceConfigId, @QueryParam("workspace-id") String workspaceId) {
         UserProfile userProfile = userManager.getRemoteUser(request);
         if (userProfile == null) {
             return Response.status(Response.Status.UNAUTHORIZED).build();
         }
-        Map<String, Object> contextMap = CoverageUiHelper.buildCoverageContextMap(projectKey, issueKey, issueId);
+        Map<String, Object> contextMap = CoverageUiHelper.buildCoverageContextMap(projectKey, issueKey, issueId, workspaceConfigId, workspaceId);
         return Response.ok(contextMap).build();
     }
+
+    @GET
+    @Path("/octane-workspaces")
+    public Response getOctaneWorkspaces(@Context HttpServletRequest request, @QueryParam("project-key") String projectKey, @QueryParam("issue-type") String issueType) {
```

Comment:
don't u need any validations on these params? for security reasons for example

Reply:
not used internally, only locally for filtering